android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java

```diff
@@ -1132,6 +1161,52 @@ public void onClick(DialogInterface dialog, int button) {
         alertDialog.show();
     }
 
+    private void permanentlyDeleteAccount() {
+        AlertDialog.Builder alert =
+                new AlertDialog.Builder(getActivity(), R.style.ThemeOverlay_BrowserUI_AlertDialog);
+        if (null == alert) {
+            return;
+        }
+        DialogInterface.OnClickListener onClickListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int button) {
+                if (button == AlertDialog.BUTTON_POSITIVE) {
+                    permanentlyDeleteAccountImpl();
+                }
+            }
+        };
+
+        AlertDialog alertDialog =
+                alert.setTitle(getResources().getString(R.string.brave_sync_delete_account_text))
+                        .setMessage(getString(R.string.brave_sync_delete_account_message))
+                        .setPositiveButton(
+                                R.string.brave_sync_delete_account_button, onClickListener)
+                        .setNegativeButton(R.string.cancel, onClickListener)
+                        .create();
+        alertDialog.getDelegate().setHandleNativeActionModesEnabled(false);
+        alertDialog.show();
+    }
+
+    private void permanentlyDeleteAccountImpl() {
+        getBraveSyncWorker().permanentlyDeleteAccount((String result) -> {
+            Log.i(TAG, "[BraveSync] permanentlyDeleteAccount result is <" + result + ">");
```

Comment:
Is it a debug log or we expect it to be useful to request from user if necessary?

Reply:
I left it here intentionally but I will remove it as will not help much on the user's side if there will be troubles.

Reply:
fixed at https://github.com/brave/brave-core/pull/13562/commits/ccd06b8f7fa54994f199efd4ab9c0e41518dceea
thanks